src/components/CarDetailComponent/CarDetailComponent.tsx

```diff
@@ -0,0 +1,58 @@
+import { Query_GetCarsQuery } from '../../api/graphql/__generated__/graphql-types';
+import { defaultImagePath } from '../../helpers/objectValues';
+import HeadingTitle from '../Text/HeadingTitle/HeadingTitle';
+import RegularText from '../Text/RegularText/RegularText';
+import {
+  Container,
+  ImageDetail,
+  InfoContainer,
+} from './CarDetailComponentStyled';
+import CarInfoItem from './CarInfoItem/CarInfoItem';
+
+interface CarDetailComponentProps {
+  carInfo: Query_GetCarsQuery;
+}
+
+const CarDetailComponent = ({ carInfo }: CarDetailComponentProps) => {
+  const {
+    batch,
+    city,
+    color,
+    condition,
+    damage_type,
+    description,
+    model,
+    odometer,
+    price,
+    sale_date,
+    title,
+    vin,
+    year,
+  } = carInfo.cars[0];

```

Comment:
why are you getting an array the cars? if this component is intented to render a single car, why don't you pass only the necessary data as prop?

Reply:
Reason: Hey gerar, well here i'm just passing the response "data" and is a type "Query_getcarQuery" and within the component i typed the prop like that and "data" has a "cars" array, 

Solution:  i will type component props as "Cars" type, so i can pass only object with data and not pass the array like i did before.